### ISSUE TITLE
Error building QT5-Tools on Windows

### ISSUE THREAD

**Issue by yhz91** [microsoft/vcpkg#8205]
The error message as shown as below:

**$ ./vcpkg.exe install qt5:x64-windows-static**
```
Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
The following packages will be built and installed:
    qt5[core]:x64-windows-static
  * qt5-tools[core]:x64-windows-static
Additional packages (*) will be modified to complete this operation.
Starting package 1/2: qt5-tools:x64-windows-static
Building package qt5-tools[core]:x64-windows-static...
-- Using cached D:/Library/vcpkg/downloads/qttools-everywhere-src-5.12.4.tar.xz
-- Using source at D:/Library/vcpkg/buildtrees/qt5-tools/src/5.12.4-00de5294bb
-- Configuring x64-windows-static-rel
-- Configuring x64-windows-static-rel done
-- Package build-x64-windows-static-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: D:/Library/vcpkg/downloads/tools/jom/jom-1.1.3/jom.exe
    Working Directory: D:/Library/vcpkg/buildtrees/qt5-tools/x64-windows-static-rel
    Error code: 2
    See logs for more information:
      D:\Library\vcpkg\buildtrees\qt5-tools\package-build-x64-windows-static-rel-out.log
      D:\Library\vcpkg\buildtrees\qt5-tools\package-build-x64-windows-static-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_qmake.cmake:32 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_build_qmake.cmake:67 (run_jom)
  installed/x64-windows-static/share/qt5/qt_build_submodule.cmake:12 (vcpkg_build_qmake)
  installed/x64-windows-static/share/qt5/qt_submodule_installation.cmake:5 (qt_build_submodule)
  ports/qt5-tools/portfile.cmake:2 (qt_submodule_installation)
  scripts/ports.cmake:91 (include)


Error: Building package qt5-tools:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-tools:x64-windows-static
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.
```
**package-build-x64-windows-static-rel-err.log**

```
jom 1.1.3 - empower your cores

jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\qtdiag\Makefile.Release [..\..\bin\qtdiag.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\qtdiag\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\pixeltool\Makefile.Release [..\..\bin\pixeltool.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\pixeltool\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\distancefieldgenerator\Makefile.Release [..\..\bin\qdistancefieldgenerator.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\distancefieldgenerator\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\qdbus\qdbusviewer\Makefile.Release [..\..\..\bin\qdbusviewer.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\qdbus\qdbusviewer\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\qdbus\Makefile [sub-qdbusviewer-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\qhelpgenerator\Makefile.Release [..\..\..\bin\qhelpgenerator.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\qhelpgenerator\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\assistant\Makefile.Release [..\..\..\bin\assistant.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\assistant\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\Makefile [sub-assistant-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\assistant\Makefile [sub-qhelpgenerator-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-assistant-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\designer\src\designer\Makefile.Release [..\..\..\..\bin\designer.exe] Error 1120
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\designer\src\designer\Makefile [release] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\designer\src\Makefile [sub-designer-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\designer\Makefile [sub-src-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-designer-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-pixeltool-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-distancefieldgenerator-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-qdbus-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\src\Makefile [sub-qtdiag-make_first] Error 2
jom: D:\Library\vcpkg\buildtrees\qt5-tools\x64-windows-static-rel\Makefile [sub-src-make_first] Error 2
```

**OS: windows10 pro**
**Visual Studio version: VS2019 Pro---16.2.5**

**Does anyone has the same problem?**  or, **How should I solve it?**

**Thanks.**

**Comment by JackBoosY** [comment]
@Neumann-A any ideas?

**Comment by Neumann-A** [comment]
Error 1120 is a linker error about missing symbols without the other logs I am unable to say anything about it.

**Comment by yhz91** [comment]
@Neumann-A 

Hi，Neumann，view the log **package-build-x64-windows-static-rel-out.log** , there is a problem with the tiff library, as follows:

```
qttoolbardialog.cpp
	link /NOLOGO /DYNAMICBASE /NXCOMPAT /INCREMENTAL:NO /SUBSYSTEM:WINDOWS "/MANIFESTDEPENDENCY:type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' publicKeyToken='6595b64144ccf1df' language='*' processorArchitecture='*'" /VERSION:5.12 /MANIFEST:embed /OUT:..\..\..\..\bin\designer.exe @C:\Users\yhz3379\AppData\Local\Temp\designer.exe.5472.5172.jom
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_code referenced in function LZMADecode
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_end referenced in function LZMACleanup
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_memusage referenced in function LZMADecode
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_lzma_preset referenced in function LZMAVSetField
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_stream_encoder referenced in function LZMAPreEncode
tiff.lib(tif_lzma.c.obj) : error LNK2019: unresolved external symbol lzma_stream_decoder referenced in function LZMADecode
..\..\..\..\bin\designer.exe : fatal error LNK1120: 6 unresolved externals
```

but I compile the x86 version, there is no problem.

Thanks.

**Comment by Neumann-A** [comment]
hmm could you tell me where the tiff dependency is hiding? Did you build some dependent port with a feature?

```
$ ./vcpkg.exe depend-info qt5-base
zlib:
bzip2:
libpng: zlib
openssl-windows:
freetype: bzip2, libpng, zlib
openssl: openssl-windows
ragel:
double-conversion:
harfbuzz[ucdn]: freetype, ragel
libjpeg-turbo:
libpq: openssl
pcre2:
sqlite3:
qt5-base: double-conversion, freetype, harfbuzz, libjpeg-turbo, libpng, libpq, openssl, pcre2, sqlite3, zlib
```

**Comment by JackBoosY** [comment]
@yhz91 I found that the latest version of qt5 in vcpkg is 5.12.5, could you update vcpkg first and rebuild qt5-tools?

Thanks.

**Comment by Neumann-A** [comment]
@JackBoosY: That will probably not solve the problem. The question here is where does the tiff dependency come from.

**Comment by Neumann-A** [comment]
@yhz91 Output of `vcpkg list` ?

**Comment by Neumann-A** [comment]
Ah ok I see we are not handling that dependency correctly. qt5-imageformats has that as a dependency and we actually need to pass the correct configure options (like qt5-base does e.g. -system-tiff and setting TIFF_LIBS). 

We probably need to go through all qt5 ports and their buildtrees and look into `config_help.txt` if we are missing some important configure switches while building the submodules.

**Comment by yhz91** [comment]
> @yhz91 Output of `vcpkg list` ?

**Comment by yhz91** [comment]
> hmm could you tell me where the tiff dependency is hiding? Did you build some dependent port with a feature?
> 
> ```
> $ ./vcpkg.exe depend-info qt5-base
> zlib:
> bzip2:
> libpng: zlib
> openssl-windows:
> freetype: bzip2, libpng, zlib
> openssl: openssl-windows
> ragel:
> double-conversion:
> harfbuzz[ucdn]: freetype, ragel
> libjpeg-turbo:
> libpq: openssl
> pcre2:
> sqlite3:
> qt5-base: double-conversion, freetype, harfbuzz, libjpeg-turbo, libpng, libpq, openssl, pcre2, sqlite3, zlib
> ```

@Neumann-A 
Hi，Neumann, Sorry, I went to dinner just now.

```
$ ./vcpkg.exe depend-info qt5-tools
zlib:
bzip2:
libpng: zlib
openssl-windows:
freetype: bzip2, libpng, zlib
openssl: openssl-windows
ragel:
double-conversion:
harfbuzz[ucdn]: freetype, ragel
libjpeg-turbo:
libpq: openssl
pcre2:
sqlite3:
qt5-base: double-conversion, freetype, harfbuzz, libjpeg-turbo, libpng, libpq, openssl, pcre2, sqlite3, zlib
qt5-declarative: qt5-base
qt5-tools: qt5-base, qt5-declarative

```

**Comment by yhz91** [comment]
> Ah ok I see we are not handling that dependency correctly. qt5-imageformats has that as a dependency and we actually need to pass the correct configure options (like qt5-base does e.g. -system-tiff and setting TIFF_LIBS).
> 
> We probably need to go through all qt5 ports and their buildtrees and look into `config_help.txt` if we are missing some important configure switches while building the submodules.

Ok，Thanks.

**Comment by yhz91** [comment]
> @yhz91 I found that the latest version of qt5 in vcpkg is 5.12.5, could you update vcpkg first and rebuild qt5-tools?
> 
> Thanks.

ok，I will update it. Thanks.

**Comment by Neumann-A** [comment]
@yhz91 the issue should be fixed with PR #8222

**Comment by chilatiaobushanghuo** [comment]
666
